KubeCon EU 2024 (#444)
BedrockStreaming/tech.bedrockstreaming.com: _posts/2024-04-09-kubecon-europe-2024.md

The presentation by Zbynek Roubalik and Jorge Turrado, KEDA maintainers, highlighted KEDA's effectiveness in meeting various load requirements through the use of a variety of external metrics, making scaling more accurate.

### Cilium

Cilium (and eBPF) were well represented at KubeCon, with a number of conferences on the subject.

This came at just the right time, as we are currently in the process of migrating to this CNI. This one, based on eBPF (with all that that implies) seemed to us to be a good replacement for our current CNI.

With cilium: no more KubeProxy, and no more IPtables, which for large clusters is a major performance factor.

Cilium also enables network traffic to be filtered with NetworkPolicies inbound and outbound.

> Another feature that we particularly like is "[cluster-mesh](https://cilium.io/use-cases/cluster-mesh/)”, which allows you to communicate between several clusters in a fairly simple way, at least on paper 😊! (conference available [⏩ here](https://www.youtube.com/watch?v=qbB3TEiOb24))

Reviewer: ```suggestion
Another feature we particularly like is "[cluster-mesh](https://cilium.io/use-cases/cluster-mesh/)”, which allows you to communicate between several clusters in a fairly simple way, at least on paper 😊! (conference available [⏩ here](https://www.youtube.com/watch?v=qbB3TEiOb24))
```

We were also able to see new features such as GatewayAPI support from version 1.15, and the use of [`stateDB`](https://github.com/cilium/statedb) from version 1.16 to manage the state of cilium objects (and therefore reconciliation in the event of inconsistent state (conference available [⏩ here](https://www.youtube.com/watch?v=YX0sql_3dt8&list=PLj6h78yzYM2PHDqy_nINY8-462uYzb18d&index=2))).

## Tools and idea we want to try 💪 

KubeCon is also a great way to discover or re-discover new technologies or tools. We won’t share an exhaustive list (see you there next year!), but here are a few we are excited about:

 * WebAssembly (especially [SpinKube](https://www.spinkube.dev/): we are running some workloads at-edge on WASM and see a huge potential there. WASM in Kubernetes is intriguing, maybe as a way to reduce pods startup time?

>  * [OpenFeature](https://openfeature.dev/): for feature-flipping, A/B testing… Glad to see an open standard emerge!

Reviewer: ```suggestion
 * [OpenFeature](https://openfeature.dev/): for feature-flipping, A/B testing… Glad to see the emergence of an open standard!
```

### Cilium as cluster-mesh

⏩ [Simplifying Multi-Cluster and Multi-Cloud Deployments with Cilium - Liz Rice, Isovalent](https://www.youtube.com/watch?v=qbB3TEiOb24)

One thing that caught our attention was the native Cluster-Mesh feature which could allow us to connect multiple clusters between them with little effort and potentially multiple cloud providers between them.

To allow each Cilium’s components to communicate between them an internal load-balancer is deployed in each cluster.

There are some prerequisites to allow cluster mesh though:

 * CIDR block must not overlapped
 * Each cluster needs to have their own unique ID and name
 * All clusters need to use Cilium as their CNI
* Cluster Mesh option has to be enabled on each cluster

> Liz Rice did a live presentation of cluster mesh capabilities with some use cases and functionalities like :

Reviewer: ```suggestion
Liz Rice did a live presentation of cluster mesh capabilities with some use cases and functionalities like:
```
